diff --git a/engines/e_ossltest.c b/engines/e_ossltest.c
index 64376247c3..70c8b62a68 100644
--- a/engines/e_ossltest.c
+++ b/engines/e_ossltest.c
@@ -319,6 +319,10 @@ static int bind_ossltest(ENGINE *e)
         return 0;
     }
 
+    ENGINE* scossl = ENGINE_by_id("symcrypt");
+    ENGINE_unregister_pkey_meths(scossl);
+    ENGINE_free(scossl);
+
     return 1;
 }
 
diff --git a/Configurations/unix-Makefile.tmpl b/Configurations/unix-Makefile.tmpl
index 68e391c088..9033c1c78d 100644
--- a/Configurations/unix-Makefile.tmpl
+++ b/Configurations/unix-Makefile.tmpl
@@ -471,6 +471,7 @@ test: tests
 {- dependmagic('tests'); -}: build_programs_nodep build_engines_nodep link-utils
 	@ : {- output_off() if $disabled{tests}; "" -}
 	( cd test; \
+	  cp $(libdir)/symcryptengine.so ../$(BLDDIR)/engines/; \
 	  mkdir -p test-runs; \
 	  SRCTOP=../$(SRCDIR) \
 	  BLDTOP=../$(BLDDIR) \
