diff --git a/engines/e_ossltest.c b/engines/e_ossltest.c
index c925d2c5a6..b1f587291c 100644
--- a/engines/e_ossltest.c
+++ b/engines/e_ossltest.c
@@ -428,6 +428,10 @@ static int bind_ossltest(ENGINE *e)
         return 0;
     }
 
+    ENGINE* scossl = ENGINE_by_id("symcrypt");
+    ENGINE_unregister_pkey_meths(scossl);
+    ENGINE_free(scossl);
+
     return 1;
 }

diff --git a/Configurations/unix-Makefile.tmpl b/Configurations/unix-Makefile.tmpl
index 18255fd43b..c7a35265dd 100644
--- a/Configurations/unix-Makefile.tmpl
+++ b/Configurations/unix-Makefile.tmpl
@@ -528,6 +528,9 @@ all: build_sw build_docs
 
 test: tests
 {- dependmagic('tests'); -}: build_programs_nodep build_modules_nodep link-utils run_tests
+	@ : {- output_off() if $disabled{tests}; "" -}
+	( cd test; \
+	  cp $(libdir)/symcryptengine.so ../$(BLDDIR)/engines/; )
 run_tests:
 	@ : {- output_off() if $disabled{tests}; "" -}
 	( SRCTOP=$(SRCDIR) \
