diff --git a/test/build.info b/test/build.info
index 6357a7f2fe..053dfa0189 100644
--- a/test/build.info
+++ b/test/build.info
@@ -14,7 +14,7 @@ IF[{- !$disabled{tests} -}]
           testutil/format_output.c testutil/tap_bio.c \
           testutil/test_cleanup.c testutil/main.c testutil/testutil_init.c \
           testutil/random.c
-  INCLUDE[libtestutil.a]=../include
+  INCLUDE[libtestutil.a]=../include ../../SymCrypt-OpenSSL/SymCryptEngine/inc
   DEPEND[libtestutil.a]=../libcrypto
 
   # Special hack for descrip.mms to include the MAIN object module
diff --git a/test/testutil/main.c b/test/testutil/main.c
index d3ccdda391..82232733c8 100644
--- a/test/testutil/main.c
+++ b/test/testutil/main.c
@@ -11,6 +11,7 @@
 #include "internal/nelem.h"
 #include "output.h"
 #include "tu_local.h"
+#include "scossl.h"
 
 #include <string.h>
 
@@ -47,6 +48,13 @@ int main(int argc, char *argv[])
 
     setup_test_framework();
 
+    if (!SCOSSL_ENGINE_Initialize())
+    {
+        test_printf_stderr("SCOSSL init failed - aborting\n");
+        return ret;
+    }
+
+
     if (setup_tests())
         ret = run_tests(argv[0]);
     cleanup_tests();
