diff --git a/test/recipes/80-test_cms.t b/test/recipes/80-test_cms.t
index 1c20ec1fd5..5fbbea4538 100644
--- a/test/recipes/80-test_cms.t
+++ b/test/recipes/80-test_cms.t
@@ -13,7 +13,7 @@ use warnings;
 use POSIX;
 use File::Spec::Functions qw/catfile/;
 use File::Compare qw/compare_text/;
-use OpenSSL::Test qw/:DEFAULT srctop_dir srctop_file with/;
+use OpenSSL::Test qw/:DEFAULT srctop_dir srctop_file/;
 use OpenSSL::Test::Utils;
 
 setup("test_cms");
@@ -27,7 +27,7 @@ my $smcont   = srctop_file("test", "smcont.txt");
 my ($no_des, $no_dh, $no_dsa, $no_ec, $no_ec2m, $no_rc2, $no_zlib)
     = disabled qw/des dh dsa ec ec2m rc2 zlib/;
 
-plan tests => 7;
+plan tests => 6;
 
 my @smime_pkcs7_tests = (
 
@@ -584,14 +584,3 @@ sub check_availability {
 
     return "";
 }
-
-# Check that we get the expected failure return code
-with({ exit_checker => sub { return shift == 6; } },
-    sub {
-        ok(run(app(['openssl', 'cms', '-encrypt',
-                    '-in', srctop_file("test", "smcont.txt"),
-                    '-stream', '-recip',
-                    srctop_file("test/smime-certs", "badrsa.pem"),
-                   ])),
-            "Check failure during BIO setup with -stream is handled correctly");
-    });
